restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -84,6 +95,209 @@ static Object translateCanonicalKeyToAlternativeKeyIfNeeded(Object canonicalKey,
     }
   }
 
+  public static DataMap fillInDefaultOnRecord(RecordDataSchema schema, DataMap dataMap)
+  {
+    DataMap dataWithDefault = new DataMap(DataMapBuilder.getOptimumHashMapCapacityFromSize(dataMap.size()));
+    dataWithDefault.putAll(dataMap);
```

Comment:
this would increase the memory consumption unnecessarily can you optimize for this. why do you need this? you're anyway traversing for list of schema fields. can we not update and return the original datamap?

Reply:
Do you mean update and return?

Reply:
yup or just update will also be fine

Reply:
I tried just update without creating new dataMap, in some cases it will return exception because the dataMap is read only.
I think in this regard we can not just assume we can update the dataMap, we have only one way to make the fill.

The exception is like the following:, it is caused by an internal field in DataMap _readOnly, which in the function we can not change it.

```
Gradle suite > Gradle test > com.linkedin.restli.internal.server.response.TestResponseUtils > testGetAbsentFieldsDefaultValues[4](case5.json) FAILED
    java.lang.AssertionError: Test failed with exception:
    java.lang.UnsupportedOperationException: Cannot mutate a read-only map
        at org.testng.Assert.fail(Assert.java:96)
        at com.linkedin.restli.internal.server.response.TestResponseUtils.testGetAbsentFieldsDefaultValues(TestResponseUtils.java:84)
```

Reply:
Ah! you may use DataComplex.clone() for shallow copy instead of manual deep copy.